wasm-test/src/lib.rs

```diff
@@ -0,0 +1,21 @@
+
+extern crate hdk;
+extern crate holochain_wasm_utils;
+
+use holochain_wasm_utils::*;
+use hdk::globals::g_mem_stack;
+
+#[no_mangle]
+pub extern "C" fn check_global_dispatch(encoded_allocation_of_input : i32) -> i32 {
+  unsafe {
+    g_mem_stack = Some(SinglePageStack::new_from_encoded(encoded_allocation_of_input as u32));
```

Comment:
Meddling just a bit above my knowledge level, but is this `u32` supposed to be the same as the `i32`s right above it?

Reply:
Yes. Wasm doesn't handle u32 explicitly but is same as i32.

Reply:
In Wasm there are `i32`, `s32` and `u32`. Most variables are `i32` which are "uninterpreted integers" since, in the end, that's how they look in memory

Reply:
So why are we using u32 here then?

Reply:
wasmi only has `i32` though.
I use `u32` because this is what we want.
Maybe we can try using `u32` in the extern function declarations?